libcontainer/specconv/spec_linux.go

```diff
@@ -407,6 +407,19 @@ func createLibcontainerMount(cwd string, m specs.Mount) (*configs.Mount, error)
 			mnt.Source = filepath.Join(cwd, m.Source)
 		}
 	}
+
+	// None of the mount arguments can contain a null byte. Normally such
+	// strings would either cause some other failure or would just be truncated
+	// when we hit the null byte, but because we serialise these strings as
+	// netlink messages (which don't have special null-byte handling) we need
+	// to block this as early as possible.
+	if strings.Contains(mnt.Destination, "\x00") ||
+		strings.Contains(mnt.Source, "\x00") ||
+		strings.Contains(mnt.Device, "\x00") ||
+		strings.Contains(mnt.Data, "\x00") {
+		return nil, errors.New("mount field contains null byte")
```

Comment:
would it be useful to include _which_ field (for users to debug what's wrong)?

In we think it's useful, we could do a loop over the fields, and check each one

Reply:
That said; it would still return the first error (unless we always check all of them), so usefulness may be limited

Reply:
This is more of a security measure, and less a mechanism to tell the user to correct something. If there's a null byte in there, something is very wrong.

Reply:
Yeah the main reason for doing this in specconv is to make sure we always reject such configurations, rather than only rejecting them when we hit the actually attack-prevention check in `bootstrapData`.

Reply:
Yes, it was mostly a nit; I don't think "security measure" and "informative/detailed" error is mutually exclusive, but this situation should be a corner-case, so not really worth spending too much time on to make it "nice".